libflux/src/ast/check/mod.rs

```diff
@@ -0,0 +1,77 @@
+use crate::ast::{walk, PropertyKey};
+use crate::ast::SourceLocation;
+use std::fmt;
+
+#[cfg(test)]
+mod tests;
+
+// check will inspect a single node and annotate it with any AST errors.
+// Unlike the ast.Check() in our Go parser, the check function in Rust
+// returns a list of errors generated during the check process PLUS
+// any errors existed before ast.check() is performed.
+pub fn check(node: walk::Node) -> Vec<Error> {
+    let mut errors = vec![];
+    walk::walk(
+        &walk::create_visitor(&mut |n| {
+            // collect any errors we found prior to ast.check.
+            for err in n.base().errors.iter() {
+                errors.push(Error {
+                    location: n.base().location.clone(),
+                    message: err.clone()
+                });
+            }
+            match n {
+                walk::Node::BadStmt(n) => {
+                    errors.push(Error {
+                        location: n.base.location.clone(),
+                        message: format!("invalid statement {}: {}",
+                                         n.base.location, n.text)
```

Comment:
Since we now store the location on the error type itself I don't think we want it in the error message. Thoughts?

Reply:
Makes sense.